src/log/MonologTarget.php

```diff
@@ -186,28 +224,11 @@ private function _createLogger(string $name): Logger
                 ->pushProcessor(new MessageProcessor());
         }
 
-        if (App::isStreamLog()) {
-            $logger->pushHandler((new StreamHandler(
-                'php://stderr',
-                Logger::WARNING,
-                bubble: false,
-            ))->setFormatter($this->formatter));
-
-            $logger->pushHandler((new StreamHandler(
-                'php://stdout',
-                $this->level,
-                bubble: false,
-            ))->setFormatter($this->formatter));
-        } else {
-            $logger->pushHandler((new RotatingFileHandler(
-                App::parseEnv(sprintf('@storage/logs/%s.log', $name)),
-                $this->maxFiles,
-                $this->level,
-                filePermission: $generalConfig->defaultFileMode,
-            ))->setFormatter($this->formatter));
-        }
+        $handlers = $this->handlers instanceof Closure
+            ? ($this->handlers)($this)
+            : $this->handlers;
```

Comment:
If `$this->handlers` is an array, we take it as-is. This makes the most intuitive sense to me, however, it _isn't_ how `\craft\log\Dispatcher::$targets` works, where they are appended to the defaults. Happy to change it either way.